File: CPP/Pelanggan.cpp
#include <bits/stdc++.h>
#include "Manusia.cpp"
#include "Pengiriman.cpp"
#include <vector>
#pragma once
using namespace std;

class Pelanggan : public Manusia {
private:
    string idPelanggan;
    string alamat;
    vector<Pengiriman> pengirimanList;  // Changed to use vector for multiple deliveries

public:
    Pelanggan(string nama = "", string nohp = "", string idPelanggan = "", string alamat = "", vector<Pengiriman> pengirimanList = {}) 
        : Manusia(nama, nohp) {
        this->idPelanggan = idPelanggan;
        this->alamat = alamat;
        this->pengirimanList = pengirimanList;
    }

    string getidPelanggan() {
        return idPelanggan;
    }

    void setidPelanggan(string idPelanggan) {
        this->idPelanggan = idPelanggan;
    }

    string getalamat() {
        return alamat;
    }

    void setalamat(string alamat) {
        this->alamat = alamat;
    }

    vector<Pengiriman> getPengiriman() {
        return pengirimanList;
    }

    void setPengiriman(vector<Pengiriman> pengirimanList) {
        this->pengirimanList = pengirimanList;
    }
};

File: Java/Pelanggan.java
import java.util.List;
import java.util.ArrayList;

public class Pelanggan extends Manusia {
    private String idPelanggan;
    private String alamat;
    private List<Pengiriman> pengirimanList;

    public Pelanggan(String nama, String nohp, String idPelanggan, String alamat, List<Pengiriman> pengirimanList) {
        super(nama, nohp);
        this.idPelanggan = idPelanggan;
        this.alamat = alamat;
        this.pengirimanList = pengirimanList != null ? pengirimanList : new ArrayList<>();
    }

    public String getIdPelanggan() {
        return idPelanggan;
    }

    public void setIdPelanggan(String idPelanggan) {
        this.idPelanggan = idPelanggan;
    }

    public String getAlamat() {
        return alamat;
    }

    public void setAlamat(String alamat) {
        this.alamat = alamat;
    }

    public List<Pengiriman> getPengiriman() {
        return pengirimanList;
    }

    public void setPengiriman(List<Pengiriman> pengirimanList) {
        this.pengirimanList = pengirimanList;
    }
}
